src/apps/reservation-list/modal/delete-reservation/delete-reservation-modal.tsx

```diff
@@ -32,12 +45,13 @@ const DeleteReservationModal: FC<DeleteReservationModalProps> = ({
   const removeSelectedReservations = () => {
     if (reservations.length > 0) {
       const reservationsToDelete = reservations
-        .map((id) => Number(isFaust(id)))
-        .filter((id) => id !== 0);
+        .filter(isPhysicalReservation)
+        .map(({ reservationIds }) => reservationIds)
+        .flat();
```

Comment:
This is they key moment of the actual change: Instead of assuming that there is one reservation id per reservation object we now have several per item. They still combine into a single array of ids to delete.